python/xorbits/core/adapter.py

```diff
@@ -143,7 +143,9 @@ def wrap_magic_method(method_name: str) -> Callable[[Any], Any]:
     def wrapped(self: DataRef, *args, **kwargs):
         # trigger on condition execution.
         mars_entity = to_mars(self)
-        if (mars_entity is None) or (
+        if hasattr(self, "to_object") and hasattr(self.to_object(), method_name):
```

Comment:
The `to_object` which will trigger execution are called twice, is it possible to optimize it?